token/services/identity/interop/htlc/deserializer.go

```diff
@@ -60,3 +52,68 @@ func (d *Deserializer) getHTLCVerifier(raw []byte) (driver.Verifier, error) {
 	v.HashInfo.HashEncoding = script.HashInfo.HashEncoding
 	return v, nil
 }
+
+func (t *TypedIdentityDeserializer) Recipients(id driver.Identity, typ string, raw []byte) ([]driver.Identity, error) {
+	if typ != htlc.ScriptType {
+		return nil, errors.New("unknown identity type")
+	}
+
+	script := &htlc.Script{}
+	err := json.Unmarshal(raw, script)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal htlc script")
+	}
+	return []driver.Identity{script.Recipient}, nil
+}
+
+func (t *TypedIdentityDeserializer) GetOwnerAuditInfo(id driver.Identity, typ string, raw []byte, p driver.AuditInfoProvider) ([][]byte, error) {
+	if typ != htlc.ScriptType {
+		return nil, errors.Errorf("invalid type, got [%s], expected [%s]", typ, htlc.ScriptType)
+	}
+	script := &htlc.Script{}
+	var err error
+	err = json.Unmarshal(raw, script)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal htlc script")
+	}
+
+	auditInfo := &ScriptInfo{}
+	auditInfo.Sender, err = p.GetAuditInfo(script.Sender)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting audit info for htlc script [%s]", id.String())
+	}
+	auditInfo.Recipient, err = p.GetAuditInfo(script.Recipient)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting audit info for script [%s]", id.String())
+	}
+
+	auditInfoRaw, err := json.Marshal(auditInfo)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed marshaling audit info for script")
+	}
+	return [][]byte{auditInfoRaw}, nil
+}
+
+type AuditDeserializer struct {
+	AuditInfoDeserializer deserializer.AuditInfoDeserializer
+}
+
+func NewAuditDeserializer(auditInfoDeserializer deserializer.AuditInfoDeserializer) *AuditDeserializer {
+	return &AuditDeserializer{AuditInfoDeserializer: auditInfoDeserializer}
+}
+
+func (a *AuditDeserializer) DeserializeAuditInfo(bytes []byte) (deserializer.AuditInfo, error) {
+	si := &ScriptInfo{}
+	err := json.Unmarshal(bytes, si)
+	if err == nil && (len(si.Sender) != 0 || len(si.Recipient) != 0) {
+		if len(si.Recipient) != 0 {
+			ai, err := a.AuditInfoDeserializer.DeserializeAuditInfo(si.Recipient)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed unamrshalling audit info [%s]", bytes)
+			}
+			return ai, nil
+		}
+		return nil, errors.Errorf("no recipient defined")
+	}
+	return nil, errors.Errorf("ivalid audit info, failed unmarshal [%s]", string(bytes))
```

Comment:
Maybe it makes sense to include also the length of sender/recipient for better debugging.

Reply:
👍